lib/websocket.c

```diff
@@ -222,6 +237,12 @@ void h2o_websocket_proceed(h2o_websocket_conn_t *conn)
         }
     } while (handled);
 
+    if (conn->_write_buf.cnt > 0) {
+        /* write */
+        h2o_socket_write(conn->sock, conn->_write_buf.bufs, conn->_write_buf.cnt, on_write_complete);
+        conn->_write_buf.cnt = 0;
```

Comment:
Reconsidering the design, could we stop resetting the counter here so that we can use `cnt` when freeing the elements?

I think that associating the value of the `cnt` _always_ with the lifetime of the elements of the buffer would lead to better understanding of the code.

Reply:
`h2o_websocket_proceed` is also called in `on_recv`, and it's public api, the user may call `h2o_websocket_proceed` more than once.

so if we do dot resetting `conn->_write_buf.cnt`,  `h2o_socket_write` maybe called more than once. this is not what we want.

Reply:
Doesn't changing the condition to `!h2o_socket_is_writing(conn->sock) && conn->_write_buf.cnt > 0` solve the issue?

Reply:
fixed